client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -81,17 +66,34 @@ class QueryComponent extends Component {
   }
 
   onChange = evt => {
-    this.setState({ selectedQuery: evt.target.value, selectedQueryKey: null });
+    this.handleEditorChange(evt);
+    this.setState({ selectedQuery: evt, selectedQueryKey: null });
   }
 
-  handleEditorChange = value => {
-    this.setState({ selectedQueryMultiline: value, selectedQueryKey: null });
-    const count = Math.min(20, Math.max(1, value.split("").filter(c => c === "\n").length));
-    this.setState({ rowHeight: `${(count + 1) * 19}px` });
+  onKeyFunction = event => {
+    const enterPressed = event.key === "Enter";
+    if (!event.shiftKey && enterPressed) {
+      this.executeQuery(event);
+    } else {
+      const evt = new Event(event.type, event);
+      event.target.dispatchEvent(evt);
+    }
+  }
+
+  onBlur = evt => {
+    this.setState({ selectedQueryHold: evt.target.value, selectedQuery: evt.target.value.replaceAll("\n", " ") });
+    this.handleEditorChange(evt.target.value.replaceAll("\n", " "));
+  }
+
+  onFocus = () => {
+    const { selectedQueryHold } = this.state;
+    this.setState({ selectedQuery: selectedQueryHold });
+    this.handleEditorChange(selectedQueryHold);
   }
 
-  handleEditorBlur = evt => {
-    this.setState({ displayEditor: false, selectedQuery: evt.target.value.replaceAll("\n", " ") });
+  handleEditorChange = value => {
+    const count = Math.min(20, Math.max(0, value.split("").filter(c => c === "\n").length));
+    this.setState({ rowHeight: `${((count + 1) * 19) + 10}px` });
```

Comment:
Can we make some of this values constants? I.e. so it's clear what they represent.

E.g. `const maxNumberOfRows = 20;` or `const lineHeight = 19;`.

Reply:
Done